src/main/java/ListUtils.java

```diff
@@ -0,0 +1,155 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class ListUtils {
+    //class to store task
+    private static class Task {
+        String description;
+        boolean isDone;
+        Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+        void markAsDone() {
+            this.isDone = true;
+        }
+
+        void markAsNotDone() {
+            this.isDone = false;
+        }
+
+        @Override
+        public String toString() {
+            return "[" + (isDone ? "X" : " ") + "] " + description;
+        }
+
+    }
+
+    private static class Todo extends Task {
+        Todo(String description) {
+            super(description);
+        }
+
+        @Override
+        public String toString() {
+            return "[T]" + super.toString();
+        }
+    }
+
+    private static class Deadline extends Task {
+
+        protected String by;
+
+        public Deadline(String description, String by) {
+            super(description);
+            this.by = by;
+        }
+        @Override
+        public String toString() {
+            return "[D]" + super.toString() + " (by: " + by + ")";
+        }
+    }
+
+    private static class Event extends Task {
```

Comment:
I really love the implementation, but do you think creating a new class would make it more cleaner rather than putting Tasks as part of List Utils